tfs/src/main/java/hudson/plugins/tfs/JenkinsEventNotifier.java

```diff
@@ -0,0 +1,126 @@
+package hudson.plugins.tfs;
+
+import hudson.plugins.tfs.model.ConnectionParameters;
+import hudson.plugins.tfs.model.JobCompletionEventArgs;
+import hudson.plugins.tfs.util.TeamRestClient;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Formatter;
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ * This class sends Jenkins events to all "connected" team collections.
+ * Current list of events:
+ * - Job Completion event
+ */
+public class JenkinsEventNotifier {
+    protected final static Logger log = Logger.getLogger(JenkinsEventNotifier.class.getName());
```

Comment:
(I don't know if we care): I think Checkstyle will flag this line for the order of the keywords:

https://stackoverflow.com/questions/10299067/modifier-keyword-order-in-java